docs/proposals/systemd-nodespec.md

```diff
@@ -0,0 +1,134 @@
+
+For Kubernetes systems, we propose the kubelet should monitor the following by default:
+
+* ```system.slice``` - for cpu and memory usage by host daemons
+* ```machine.slice``` - for baremetal scenarios that may have super privileged containers or other vms on the node machine.
+
+It is assumed that ```kubelet``` and ```docker``` are launched via units in the ```system.slice```.
+
+The ```kubelet``` should detect the unit that has launched its service, and should discover the unit that manages the ```docker``` daemon.
+
+## Unified cgroup hierarchy
+
+Classical cgroup hierarchies were orthogonal per resource group controller.  For example, you could have one hierarchy for ```/sys/fs/cgroup/cpu/...``` and a completely different hierarchy for ```/sys/fs/cgroup/cpuacct``` even though one depended on the other.
+
+The direction is to move toward a single unified cgroup hierarchy, where the per-controller hierarchies are eliminated in favor of 
+just ```/sys/fs/cgroup/```.
+
+In this environment, there is automatic propagation of controllers, processes may only exist on leaf nodes in the hierarchy.
+
+```systemd``` has laid the ground work for enforcing a single unified control group hierarchy.
+
+It implicitly enforces a unified hierarchy by virtue of its representation of ```slice```, ```scope```, and ```service``` since a service
+only belongs to a single parent branch in the ```slice``` hierarchy.
+
+To align with the unified hierarchy, this document proposes the following set of ```slice``` definitions:
+
+* ```kubelet-besteffort.slice``` - cgroup that holds user containers with best effort quality of service
+* ```kubelet-burstable.slice``` - cgroup that holds user containers that have burstable quality of service
+* ```kubelet-burstable-guaranteed.slice```- cgroup that holds user containers that have guaranteed quality of service
+
+## cgroup single writer (with delegation)
+
+A major motivation towards moving towards a unified hierarchy is having a single writer for changes to the ```/sys/fs/cgroup``` hierarchy.
+
+In a ```systemd``` environment, direct writes to the ```cgroup``` tree should not be made by other systems unless being explicitly delegated.  The only delegation proposed in this document is that the ```docker``` daemon is instructed the parent ```cgroup``` to launch
+the user container based on the containers quality of service.
+
+The ```kubelet``` itself should not directly modify the ```cgroup``` hierarchy when it detects that its run via ```systemd```.
+
+Instead, on start-up, the ```kubelet``` should interface with ```systemd``` to provision ```slices``` for each quality of service.
+
+Any resource constraints that are imposed on the ```kubelet``` or ```docker``` daemons should happen via manipulation of the unit file
+properties via the ```systemd``` API.  Unit file changes for resource controls are applied automatically on write so no restart is needed.
```

Comment:
I was planning on using set property calls which can specify just being runtime only or it can specify to carry across reboots.  Will clarify the doc.